### PR TITLE
drivers: intel_gna: Intel GNA Neural Network Accelerator Driver

### PR DESCRIPTION
**Pull request by sathishkuttan**
drivers: intel_gna: Intel GNA Neural Network Accelerator Driver

This PR adds APIs and a driver implementation for the Intel GNA neural network accelerator in Intel S1000 ASIC.

### CODE REVIEW

File: include/gna.h [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+	gna_api_config		configure;
+	gna_api_register	register_model;
+	gna_api_deregister	deregister_model;
+	gna_api_infer		infer;
+};
+
+/**
+ * @endcond
+ */
+
```
Comment (nashif) [inline include/gna.h]: You will need to document the API with doxygen decorations.

**Review by nashif — CHANGES_REQUESTED** [review]

File: include/gna.h (hunk 1 above)
Comment (sathishkuttan) [reply include/gna.h]: Good catch. Done.

File: boards/xtensa/intel_s1000_crb/intel_s1000_crb_defconfig [hunk 2]
```diff
@@ -54,3 +54,6 @@ CONFIG_AUDIO_CODEC=y
 CONFIG_AUDIO_TLV320DAC=y
 CONFIG_AUDIO_DMIC=y
 CONFIG_AUDIO_INTEL_DMIC=y
+
+CONFIG_NEURAL_NET=y
```
Comment (pfalcon) [inline boards/xtensa/intel_s1000_crb/intel_s1000_crb_defconfig]: IMHO, that's too generic. If this is an option to enable neural net **accelerator** **drivers**, then the most important thing about it is that it's *hardware driver*, and that's completely not mirrored in the option name.

Comment (sathishkuttan) [reply boards/xtensa/intel_s1000_crb/intel_s1000_crb_defconfig]: How about `CONFIG_NEURAL_NET_ACCEL`

**Review by nashif — APPROVED** [review]

File: boards/xtensa/intel_s1000_crb/intel_s1000_crb_defconfig (hunk 2 above)
Comment (pfalcon) [reply boards/xtensa/intel_s1000_crb/intel_s1000_crb_defconfig]: Definitely better, +1 from me.

Comment (sathishkuttan) [reply boards/xtensa/intel_s1000_crb/intel_s1000_crb_defconfig]: Thanks. Updated to `CONFIG_NEURAL_NET_ACCEL`.

Comment (pfalcon) [reply boards/xtensa/intel_s1000_crb/intel_s1000_crb_defconfig]: Thanks too! So, I'm not qualified to review the rest of the patch to approve it, but informally, +1, it's great to see such cool features added to Zephyr.

**Pull request merged**